assets/main.js

```diff
@@ -26,3 +31,56 @@ document.addEventListener('DOMContentLoaded', () => {
         }
     });
 });
+
+function startEventSource() {
+	if (queryStreamEvent) return;
+
+	if (!window.location.pathname.startsWith("/querylog")) return;
+
+	const query_logs = document.getElementById("query-logs");
+
+	queryStreamEvent = new EventSource(`/api/queries`);
+
+	queryStreamEvent.onmessage = ({ data }) => {
+		try {
+		    const payload = JSON.parse(data);
+                    let row = query_logs.insertRow(-1);
+                    let c1 = row.insertCell(0);
+                    c1.innerText = `${payload.fin}`;
+                    let c2 = row.insertCell(1);
+                    c2.innerText = `${payload.typ}`;
+                    let c3 = row.insertCell(2);
+                    c3.innerText = `${payload.domain}`;
+                    let c4 = row.insertCell(3);
+                    c4.innerText = `${payload.client}`;
+                    let c5 = row.insertCell(4);
+                    c5.innerText = `${payload.rcode}`;
+                    let c6 = row.insertCell(5);
+                    c6.innerText = `${payload.time_taken} ms`;
+                    let c7 = row.insertCell(6);
+                    c7.innerText = `${payload.status}`;
+                    let c8 = row.insertCell(7);
+                    c8.innerText = "BLOCK (fake)";
```

Comment:
I'd appreciate how to get this "BLOCK" button as it is in the original table... do I need to put the full HTML here? help is appreciated.